include/maths/CBoostedTreeImpl.h

```diff
@@ -472,85 +424,79 @@ class MATHS_EXPORT CBoostedTreeImpl final {
     void startProgressMonitoringTrainIncremental();
 
     //! Record the training state using the \p recordTrainState callback function
-    void recordState(const TTrainingStateCallback& recordTrainState) const;
+    void outputState(const TTrainingStateCallback& recordTrainState) const;
```

Comment:
"output" is out of sync with the documentation and the parameter name.

Reply:
For me it helps readability to make it clearer that this is sending stuff externally. We also for example capture the best hyperparameters and to me recording could just as easily be internally capturing the stuff rather than writing it elsewhere. I agree that this is inconsistent as is and I don't feel too strongly about the output terminology. It might be too much to make the knock on changes for consistency, I'll see, in which case I'll revert.